### ISSUE TITLE
fluxctl bash completion?

### ISSUE THREAD

**Issue by johnraz** [fluxcd/flux#1593]
Hello,

As fluxctl seems to be leveraging cobra, it should be quite easy to achieve by making use of this cobra feature: https://github.com/eparis/cobra/blob/master/bash_completions.md

I would be happy to implement it but at the same time I'd like to have some feedback from the team before starting.

Would you be interested into getting bash completion setup for fluxctl?

Thanks!

**Comment by squaremo** [comment]
Sure, great idea! Even basic completion (i.e.,, not going looking for nouns) would be helpful. As part of this, it'd be nice to have some instructions for setting it up when installing fluxctl, too.

**Comment by johnraz** [comment]
Cool, I will get this going then 👍

**Comment by jrob** [comment]
Any word?

**Comment by yiannistri** [comment]
Hey, I've just opened a PR for this. I still need to figure how to put these in the right path automatically when installing from a package manager i.e. brew.

**Comment by yiannistri** [comment]
For brew we need to amend the fluxctl formula to include [these](https://github.com/Homebrew/homebrew-core/blob/master/Formula/kubernetes-cli.rb#L32-L38) in order to generate it at install time. I suspect other package managers will need something similar. I can work on a PR after this issue is addressed.

**Comment by dholbach** [comment]
> For brew we need to amend the fluxctl formula to include [these](https://github.com/Homebrew/homebrew-core/blob/master/Formula/kubernetes-cli.rb#L32-L38) in order to generate it at install time. I suspect other package managers will need something similar. I can work on a PR after this issue is addressed.

Nice one, @yiannistri - let me know how you want to proceed. It'd be great to add these in, homebrew, snap, chocolatey - where possible.

**Comment by hiddeco** [comment]
Do we want to keep the issue open until this has made it to the package managers?

**Comment by dholbach** [comment]
I think we can just go and file follow-up PRs.

**Comment by yiannistri** [comment]
@dholbach I'm more than happy to help out with these. Shall I start with brew?

**Comment by dholbach** [comment]
That sounds great Yiannis!
